main/src/hadd.cxx

```diff
@@ -162,6 +168,33 @@ int main( int argc, char **argv )
       } else if ( strcmp(argv[a],"-O") == 0 ) {
          reoptimize = kTRUE;
          ++ffirst;
+      } else if ( strcmp(argv[a],"-j") == 0 ) {
+         // If the number of processes is not specified, use the default.
+         if (a+1 != argc && argv[a+1][0] != '-') {
+            // number of processes specified
+            Long_t request = 1;
+            for (char *c = argv[a+1]; *c != '\0'; ++c) {
+               if (!isdigit(*c)) {
+                  // Wrong number of Processes. Use the default:
+                  std::cerr << "Error: could not parse the number of processes to run in parallel passed after -j: " << argv[a+1] << ". We will use the system maximum.\n";
+                  request = 0;
+                  break;
+               }
+            }
+            if (request == 1) {
+               request = strtol(argv[a+1], 0, 10);
+               if (request < kMaxLong && request >= 0) {
```

Comment:
Not necessarily.

From cpp reference:
-  If successful, an integer value corresponding to the contents of str is returned.
- If the converted value falls out of range of corresponding return type, a range error occurs (setting errno to ERANGE) and LONG_MAX, LONG_MIN, LLONG_MAX or LLONG_MIN is returned.
- If no conversion can be performed, ​0​ is returned

Which means that this may be incorrect when no conversion can be performed. There are a couple more places were hadd was already doing that. I'll check.